pkg/controller/pxc/controller_test.go

```diff
@@ -540,3 +541,749 @@ var _ = Describe("Finalizer delete-pxc-pvc", Ordered, func() {
 		})
 	})
 })
+
+var _ = Describe("Authentication policy", Ordered, func() {
+	ctx := context.Background()
+
+	const ns = "auth-policy"
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+	})
+
+	AfterAll(func() {
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context("Cluster is deployed with ProxySQL", Ordered, func() {
+		const crName = "auth-policy-proxysql"
+		crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+		cr, err := readDefaultCR(crName, ns)
+		It("should read default cr.yaml", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should create PerconaXtraDBCluster", func() {
+			cr.Spec.HAProxy.Enabled = false
+			cr.Spec.ProxySQL.Enabled = true
+
+			Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+		})
+
+		It("should reconcile", func() {
+			_, err := reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should use mysql_native_password", func() {
+			sts := appsv1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      crName + "-pxc",
+					Namespace: ns,
+				},
+			}
+			err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&sts), &sts)
+			Expect(err).NotTo(HaveOccurred())
+
+			envFound := false
+			for _, c := range sts.Spec.Template.Spec.Containers {
```

Comment:
There is a better way to assert things like this, with `github.com/onsi/gomega/gstruct`.
Here is an example how to do it: [ Expect(sts.Spec.Template.Spec.Containers).Should(](https://github.com/percona/percona-server-mysql-operator/blob/b5aa84db349c8d1d8d3150ebe45cccae979142d9/pkg/controller/ps/controller_test.go#L155)

Reply:
IMHO previous version was much more clear but https://github.com/percona/percona-xtradb-cluster-operator/pull/1415/commits/fe788b6b8ffc67beebb2f311183a59fd72ab463f